### DISCUSSION TITLE
Using language with script and CDN

### DISCUSSION THREAD

**Discussion by guenlm** [handsontable/hyperformula#1340] [API Questions]
Hello

If I use <script> and cdn for hyperformula, how I can import language because this code doesn't work..
HTML : 
`<script src="https://cdn.jsdelivr.net/npm/hyperformula@2.6.0/dist/languages/frFR.js"></script>`

js :
```
import frFR from 'hyperformula/es/i18n/languages/frFR';
HyperFormula.registerLanguage('frFR', frFR);
```

Thank's

**Comment by guenlm** [comment]
ok I found the solution for JS with a CDN (in the documentation !)
`const frFR = HyperFormula.languages.frFR;`

but I have the same problem in a nodejs project...

`import { HyperFormula } from 'hyperformula';
import frFR from 'hyperformula/es/i18n/languages/frFR';`

ok for HF but not for the languages...
I already put  in my package.json
`"type": "module",`

thank's for your help...
guen

**Comment by sequba** [comment]
Hi @guenlm, thanks for reaching out to us. 

Take a look at this demo: https://codesandbox.io/s/github/handsontable/hyperformula-demos/tree/2.6.x/i18n?file=/src/hyperformulaConfig.js:41-96

If it doesn't help, please, share the code that produces the issue via https://codesandbox.io/

**Comment by guenlm** [comment]
thank's but the problem is with nodejs...
sandbox here
<https://codesandbox.io/p/devbox/node-js-k8dsq1?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clp739dly0007356htnxj1744%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp739dly0003356hwsuxw578%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp739dly0005356hd5zswary%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clp739dly0006356hzhwymuxz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clp739dly0003356hwsuxw578%2522%253A%257B%2522id%2522%253A%2522clp739dly0003356hwsuxw578%2522%252C%2522activeTabId%2522%253A%2522clp73e88k00ln356j2ocpuqqn%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522id%2522%253A%2522clp73b9rs001h356jafkuva1k%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522id%2522%253A%2522clp73e88k00ln356j2ocpuqqn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%257D%252C%2522clp739dly0006356hzhwymuxz%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clp739dly0006356hzhwymuxz%2522%257D%252C%2522clp739dly0005356hd5zswary%2522%253A%257B%2522id%2522%253A%2522clp739dly0005356hd5zswary%2522%252C%2522activeTabId%2522%253A%2522clp73efdn00mo356jxwz5ifsl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp739dly0004356hp3knzr89%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522id%2522%253A%2522clp73efdn00mo356jxwz5ifsl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clp73ef8w00nqedg6cdsu5y4j%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D>
:
guen

Le lun. 20 nov. 2023 à 10:53, Kuba Sekowski ***@***.***> a
écrit :

**Comment by sequba** [comment]
Are you sure this link is correct? This looks like an empty app...

**Comment by guenlm** [comment]
may be a problem with permissions.. i don't have pro subscription for
sandbox...
[image: image.png]

i can add you with private email...
[image: image.png]

Le lun. 20 nov. 2023 à 18:32, Kuba Sekowski ***@***.***> a
écrit :

**Comment by guenlm** [comment]
another try :
[https://codesandbox.io/p/devbox/test-hyperformula-with-nodejs-pgkwd8?file=%2Fpackage.json&layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clp739dly0007356htnxj1744%2522%252C%2522sizes%2522%253A%255B70%252C30%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clp739dly0003356hwsuxw578%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clp739dly0005356hd5zswary%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clp739dly0006356hzhwymuxz%2522%257D%255D%252C%2522sizes%2522%253A%255B100%255D%257D%255D%252C%2522sizes%2522%253A%255B100%252C0%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clp739dly0003356hwsuxw578%2522%253A%257B%2522id%2522%253A%2522clp739dly0003356hwsuxw578%2522%252C%2522tabs%2522%253A%255B%257B%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Findex.js%2522%252C%2522id%2522%253A%2522clp73e88k00ln356j2ocpuqqn%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clp8inqhd011h356hoiumohk8%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fpackage.json%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522activeTabId%2522%253A%2522clp8inqhd011h356hoiumohk8%2522%257D%252C%2522clp739dly0006356hzhwymuxz%2522%253A%257B%2522id%2522%253A%2522clp739dly0006356hzhwymuxz%2522%252C%2522tabs%2522%253A%255B%255D%257D%252C%2522clp739dly0005356hd5zswary%2522%253A%257B%2522id%2522%253A%2522clp739dly0005356hd5zswary%2522%252C%2522activeTabId%2522%253A%2522clp73efdn00mo356jxwz5ifsl%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clp739dly0004356hp3knzr89%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TASK_LOG%2522%252C%2522taskId%2522%253A%2522start%2522%257D%252C%257B%2522id%2522%253A%2522clp73efdn00mo356jxwz5ifsl%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522TERMINAL%2522%252C%2522shellId%2522%253A%2522clp73ef8w00nqedg6cdsu5y4j%2522%257D%255D%257D%257D%252C%2522showDevtools%2522%253Afalse%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D](url)

**Reply by sequba** [reply]
This link works. Our team will analyze the situation and get back to you shortly.

**Comment by guenlm** [comment]
Ok, thanks.

Le mer. 22 nov. 2023 à 14:36, Kuba Sekowski ***@***.***> a
écrit :

**Comment by sequba** [comment]
@guenlm After analyzing this issue, I can confirm there is a bug in our configuration. The language files are not compatible with the ES module system in Node. It will be fixed in one of the upcoming releases. I created an issue https://github.com/handsontable/hyperformula/issues/1344 to track it.

In the meantime, you may try using the CommonJS module system as a workaround:
```js
const { HyperFormula } = require('hyperformula');
const frFR = require('hyperformula/commonjs/i18n/languages/frFR');
```

**Comment by AMBudnik** [comment]
Hi @guenlm, The following issue linked by @sequba is fixed and was just released in the HyperFormula v3.0.0 version. Here  https://hyperformula.handsontable.com/guide/release-notes.html#_3-0-0 you can check the full changelog. I will close this issue as solved. If you happen to find any issues please let us know in the comments.
